pkg/controller/common/controller_context.go

```diff
@@ -42,6 +44,7 @@ type ControllerContext struct {
 	APIExtInformerFactory                               apiextinformers.SharedInformerFactory
 	ConfigInformerFactory                               configinformers.SharedInformerFactory
 	OperatorInformerFactory                             operatorinformers.SharedInformerFactory
+	EtcdInformer                                        operatorv1.EtcdInformer
```

Comment:
Any reason you can't use the `OperatorInformerFactory` to grab the etcd informer where needed?

Reply:
I am using the OperatorInformerFactor to grab the etcd Informer https://github.com/openshift/machine-config-operator/pull/1288/files#diff-c0b7dc759bd9a095fee62f9c075d7edbR121

Since all the other informers like the [`InfraInformer`](https://github.com/openshift/machine-config-operator/blob/edee7a77f86bfe27018d8353cf32f7f8492a5a07/pkg/operator/operator.go#L126) were coming in as an argument the constructor, I used the same for consistency.

Reply:
The `InfraInformer` uses the `ConfigInformerFactory` to instantiate here: https://github.com/openshift/machine-config-operator/blob/4838fbc6e7fc32a22913fe243f99bdd1993e8d83/cmd/machine-config-operator/start.go#L69

You can use the `OperatorInformerFactory` in a similar way for etcd and then you don't need any modifications to the controller_context stuff.